File: backend/app/services/incident_service.py
from sqlalchemy.ext.asyncio import AsyncSession
from typing import List, Optional
from fastapi import HTTPException
from app.models.incident import Incident
from app.schemas.incident import IncidentCreate, IncidentUpdate
from app.repositories.incident_repository import IncidentRepository

class IncidentService:
    def __init__(self, session: AsyncSession):
        self.repository = IncidentRepository(session)

    async def create_incident(self, incident_in: IncidentCreate) -> Incident:
        return await self.repository.create(incident_in)

    async def get_incident(self, incident_id: str) -> Incident:
        incident = await self.repository.get(incident_id)
        if not incident:
            raise HTTPException(status_code=404, detail="Incident not found")
        return incident

    async def get_incidents(self, skip: int = 0, limit: int = 100) -> List[Incident]:
        return await self.repository.get_multi(skip=skip, limit=limit)

    async def update_incident(self, incident_id: str, incident_in: IncidentUpdate) -> Incident:
        db_incident = await self.repository.get(incident_id)
        if not db_incident:
            raise HTTPException(status_code=404, detail="Incident not found")
        return await self.repository.update(db_incident, incident_in)

    async def delete_incident(self, incident_id: str) -> None:
        db_incident = await self.repository.get(incident_id)
        if not db_incident:
            raise HTTPException(status_code=404, detail="Incident not found")
        await self.repository.delete(db_incident)

File: frontend/src/services/incidentService.ts
import api from './api';
import type { Incident, IncidentCreate, IncidentUpdate } from '../types';

export const incidentService = {
  getIncidents: async (skip: number = 0, limit: number = 100): Promise<Incident[]> => {
    const response = await api.get('/incidents', { params: { skip, limit } });
    return response.data;
  },

  getIncident: async (id: string): Promise<Incident> => {
    const response = await api.get(`/incidents/${id}`);
    return response.data;
  },

  createIncident: async (data: IncidentCreate): Promise<Incident> => {
    const response = await api.post('/incidents', data);
    return response.data;
  },

  updateIncident: async (id: string, data: IncidentUpdate): Promise<Incident> => {
    const response = await api.patch(`/incidents/${id}`, data);
    return response.data;
  },

  deleteIncident: async (id: string): Promise<void> => {
    await api.delete(`/incidents/${id}`);
  },

  getHealth: async (): Promise<{ status: string, service: string }> => {
    const response = await api.get('/health');
    return response.data;
  }
};
